### ISSUE TITLE
Verification dependent on other fields

### ISSUE THREAD

**Issue by GregDThomas** [tonybaines/gestalt#5]
The Bean Validation JSR allows simple `NotNull`, `Min`, etc. validation on a per-element basis. However, it would be useful to provide an interface level validation. For example, with when configuring a proxy, you may have something like;

```
proxy-enabled = true
proxy-host = proxy.domain.com
proxy-port = 8080
```

You only want to validate that proxy.host and proxy.port are present only if proxy-enabled is true. Note that this could be handled by making the proxy configuration a sub interface, e.g.

```
interface Config {
    ProxyConfig getProxyConfigration();
}

interface ProxyConfig {
    @NotNull
    String getHostName();
    @NotNull
    int getPort();
}
```
however this prevents the storage of the proxy details via the serialisation functionality if the the proxy is disabled - so it's can't simply be enabled/disabled by use of a single flag.

As a thought, this could be done using an annotation similar to;
@NotNullIf(proxy-enabled=true)

**Comment by tonybaines** [comment]
Perhaps an annotation at the interface level on the lines of

```java
@ValidationRule("if(enabled==true) (hostName != null && port != null")
interface ProxyConfig { ... }
```
The trick is getting the scope right for evaluating the rule

**Comment by tonybaines** [comment]
With Java 8 default methods this looks a lot easier

**Comment by tonybaines** [comment]
Fixed in 2.0.1
